### PR TITLE
Add headless client component

### PR DESCRIPTION
**Pull request by Glowbal**
Original work for ACE3 by @jonpas

### CODE REVIEW

**Comment by Glowbal** [comment]
Original pull request: acemod/ACE3#3107

File: addons/headless/XEH_postInit.sqf [hunk 1]
```diff
@@ -11,6 +11,6 @@ if (hasInterface && !isServer) exitWith {};
         };
     } else {
         // Register HC (this part happens on HC only)
-        ["ACE_HeadlessClientJoined", [player]] call EFUNC(common,globalEvent);
+        ["ACE_HeadlessClientJoined", [player]] call (ace_common_globalEvent);
```
Comment (bux578) [inline addons/headless/XEH_postInit.sqf]: It should be `ace_common_fnc_globalEvent`.
And I guess you can ommit the parantheses.

File: addons/headless/XEH_postInit.sqf [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+["SettingsInitialized", {
+    if (isServer) then {
+        // Add disconnect EH if HC transferring enabled
+        if (GVAR(Enabled)) then {
+            addMissionEventHandler ["HandleDisconnect", {_this call FUNC(handleDisconnect)}];
+        };
+    } else {
+        // Register HC (this part happens on HC only)
+        ["ACE_HeadlessClientJoined", [player]] call (ace_common_globalEvent);
```
Comment (bux578) [inline addons/headless/XEH_postInit.sqf]: `ace_common_fnc_globalEvent`

File: addons/headless/XEH_postInit.sqf [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    };
+}] call (ace_common_addEventHandler);
```
Comment (bux578) [inline addons/headless/XEH_postInit.sqf]: `ace_common_fnc_addEventHandler`

File: addons/headless/XEH_preInit.sqf [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+PREP(handleDisconnect);
+PREP(handleInitPost);
+PREP(moduleInit);
+PREP(rebalance);
+PREP(transferGroups);
+
+if (isServer) then {
+    GVAR(headlessClients) = [];
+    GVAR(inRebalance) = false;
+    ["ACE_HeadlessClientJoined", FUNC(handleConnectHC)] call (ace_common_addEventHandler);
```
Comment (bux578) [inline addons/headless/XEH_preInit.sqf]: `ace_common_fnc_addEventHandler`

File: addons/headless/functions/fnc_moduleInit.sqf [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+ */
+#include "script_component.hpp"
+
+if (!isServer) exitWith {};
+
+params ["_logic", "", "_activated"];
+
+if (!_activated) exitWith {};
+
+[_logic, QGVAR(Enabled), "Enabled"] call (ace_common_readSettingFromModule);
```
Comment (bux578) [inline addons/headless/functions/fnc_moduleInit.sqf]: `ace_common_fnc_readSettingFromModule`

File: addons/headless/functions/fnc_rebalance.sqf [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+params ["_force"];
+
+TRACE_3("Rebalance",GVAR(inRebalance),GVAR(headlessClients),_force);
+
+// Exit if waiting for rebalance or no HCs present
+if (GVAR(inRebalance) || {GVAR(headlessClients) isEqualTo []}) exitWith {};
+
+// Transfer after rebalance delay
+[FUNC(transferGroups), [_force], GVAR(Delay)] call (ace_common_waitAndExecute);
```
Comment (bux578) [inline addons/headless/functions/fnc_rebalance.sqf]: `ace_common_fnc_waitAndExecute`

**Pull request merged**